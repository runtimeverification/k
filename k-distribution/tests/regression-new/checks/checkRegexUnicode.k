// Copyright (c) Runtime Verification, Inc. All Rights Reserved.
module CHECKREGEXUNICODE
syntax lexical Foo = r"[^a😊][😦-ab-🙁]"
syntax Bar ::= r"[^a😊][😦-ab-🙁]" [token]
endmodule
