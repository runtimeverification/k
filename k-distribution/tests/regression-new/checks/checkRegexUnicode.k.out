[Error] Outer Parser: Invalid character range '😦-a'. Start of range U+1F626 is greater than end of range U+0061.
	Source(checkRegexUnicode.k)
	Location(3,1,3,41)
	3 |	syntax lexical Foo = r"[^a😊][😦-ab-🙁]"
	  .	^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Outer Parser: Unsupported non-ASCII characters found in character class range: [😦, 🙁]
	Source(checkRegexUnicode.k)
	Location(3,1,3,41)
	3 |	syntax lexical Foo = r"[^a😊][😦-ab-🙁]"
	  .	^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Outer Parser: Unsupported non-ASCII characters found in negated character class: [😊]
	Source(checkRegexUnicode.k)
	Location(3,1,3,41)
	3 |	syntax lexical Foo = r"[^a😊][😦-ab-🙁]"
	  .	^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Outer Parser: Invalid character range '😦-a'. Start of range U+1F626 is greater than end of range U+0061.
	Source(checkRegexUnicode.k)
	Location(4,16,4,43)
	4 |	syntax Bar ::= r"[^a😊][😦-ab-🙁]" [token]
	  .	               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Outer Parser: Unsupported non-ASCII characters found in character class range: [😦, 🙁]
	Source(checkRegexUnicode.k)
	Location(4,16,4,43)
	4 |	syntax Bar ::= r"[^a😊][😦-ab-🙁]" [token]
	  .	               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Outer Parser: Unsupported non-ASCII characters found in negated character class: [😊]
	Source(checkRegexUnicode.k)
	Location(4,16,4,43)
	4 |	syntax Bar ::= r"[^a😊][😦-ab-🙁]" [token]
	  .	               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Compiler: Had 6 parsing errors.
